great_expectations/cli/datasource.py

```diff
@@ -608,36 +580,54 @@ def _collect_redshift_credentials(default_credentials={}):
 
     return credentials
 
-def _add_spark_datasource(context, prompt_for_datasource_name=True):
-    path = click.prompt(
-        msg_prompt_filesys_enter_base_path,
-        # default='/data/',
-        type=click.Path(
-            exists=True,
-            file_okay=False,
-            dir_okay=True,
-            readable=True
-        ),
-        show_default=True
-    )
-    if path.startswith("./"):
-        path = path[2:]
+def _add_spark_datasource(context, passthrough_generator_only=True, prompt_for_datasource_name=True):
+    if not load_library("pyspark"):
+        return None
 
-    if path.endswith("/"):
-        path = path[:-1]
+    if passthrough_generator_only:
+        datasource_name = "files_spark_datasource"
 
-    datasource_name = os.path.basename(path) + "__dir"
-    if prompt_for_datasource_name:
-        datasource_name = click.prompt(
-            msg_prompt_datasource_name,
-            default=datasource_name,
+        configuration = SparkDFDatasource.build_configuration(generators={
+            "default": {
+                "class_name": "PassthroughGenerator",
+            }
+        }
+        )
+
+    else:
+        path = click.prompt(
+            msg_prompt_filesys_enter_base_path,
+            # default='/data/',
+            type=click.Path(
+                exists=True,
+                file_okay=False,
+                dir_okay=True,
+                readable=True
+            ),
             show_default=True
         )
+        if path.startswith("./"):
+            path = path[2:]
```

Comment:
This would be much more robust across platforms (looking at you window)  and simple with a call to os.path.basename (IIRC) or similar

Reply:
Good find! Addressed.